src/Core/Managed/Shared/Channel/InMemoryChannel.cs

```diff
@@ -133,7 +143,14 @@ public void Flush()
         /// <param name="timeout">Timeout interval to abort sending.</param>
         public void Flush(TimeSpan timeout)
         {
-            this.transmitter.Flush(timeout);
+            if (!isDisposed)
+            {
+                this.transmitter.Flush(timeout);
+            }
+            else
+            {
+                throw new ObjectDisposedException("InMemoryChannel", "The telemetry channel has been disposed.");
```

Comment:
It sounds to me like what you are describing is that none of these classes should be disposable.
But they are already disposable.
I've fixed the issue with lost telemetry occurring because flush was not called on dispose.
The existing code behavior already stopped the runner thread when dispose was called and anything sent after that was just left in memory.  You are correct that a user could have kept using the object after disposing of it but that "scenario" is wrong in my opinion.
I object to the idea that a disposable object, which has been disposed of, should keep working after it was disposed of.  
Maybe some user will post an issue on a forum complaining that they disposed of the object and then got an exception when they tried to keep using it but at that point the whole .NET community should be expected to reply that this is expected.
I don't believe we should hide user bugs, we don't provide any value to the user by silently ignoring their mistakes and pretending their code is fine when it isn't.